lib/mysql.js

```diff
@@ -249,6 +240,28 @@ MySQL.prototype.updateOrCreate = MySQL.prototype.save =
       callback(err, data, meta);
     });
   };
+  
+/**
+ * Replace if the model instance exists with the same id or create a new instance
+ *
+ * @param {String} model The model name
+ * @param {Object} data The model instance data
+ * @param {Function} [callback] The callback function
+ */
+MySQL.prototype.replaceOrCreate = function (model, data, options, callback) {
+  this.modifyOrCreate(model, data, options, 'buildReplaceFields', callback);
```

Comment:
@bajtos You proposed to use something like `this.buildReplaceFields` here; however the `buildReplaceFields` has `this` keyword in it; and it breaks `var props = this.getModelDefinition(model).properties;` inside `buildReplaceFields`; how about passing the function name like here and handeling it in `modifyOrCreate`?